yapapi/runner/__init__.py

```diff
@@ -117,7 +117,10 @@ async def decorate_demand(self, demand: DemandBuilder) -> None:
         demand.ensure(f"({com.PRICE_MODEL}={com.PriceModel.LINEAR.value})")
 
     async def score_offer(self, offer: rest.market.OfferProposal) -> float:
-        linear: com.ComLinear = com.ComLinear.from_props(offer.props)
+        try:
+            linear: com.ComLinear = com.ComLinear.from_props(offer.props)
+        except Exception:
```

Comment:
What will happen if there for example `NameError` inside this `try..except`? It looks too general to just return `SCORE_REJECTED`. Maybe if it would also log the real traceback...

Reply:
I agree with @jivan that we loose information here if we just catch all exceptions and return `SCORE_REJECTED`. It's fine to catch some specific error types, but let the unexpected ones bubble up to where they can be handled/reported properly.

Reply:
Also note that `score_offer()` is really client code -- it's part of the strategy that is passed as an argument to `Engine.map()`, so we should expect it to raise exceptions.

Reply:
> Also note that `score_offer()` is really client code... so we should expect it to raise exceptions.
nice catch

Reply:
> Also note that `score_offer()` is really client code -- it's part of the strategy that is passed as an argument to `Engine.map()`, so we should expect it to raise exceptions.

Ok, but we can ask developers implementing MarketStrategy not to raise exceptions, but to reject invalid offers (i.e. return `SCORE_REJECTED`). For example, in C++ you can use `noexcept` keyword. AFAIK Python type hinting does not allow specifying this, but maybe there is some other solution?

Reply:
Asking developers not to raise exceptions is like asking them to write correct code...
Do we want to hang SDK in case of error in client implementation?

Reply:
> Asking developers not to raise exceptions is like asking them to write correct code...
> Do we want to hang SDK in case of error in client implementation?

That's why I want to catch them (https://github.com/golemfactory/yapapi/pull/32#discussion_r490202273). But in other languages you can enforce no exceptions and I wondered if it's possible to achieve this using type hinting or something similar.

Reply:
Guys, as I've asked @ todays dev standup, pls let code that is compatible with decentralized market into master, and continue those advanced (a bit nitpicked tbh) considerations to other issues / PRs. We need to have master working with our solution to do the DEMO.